### ISSUE TITLE
Game crashes on startup on Linux

### ISSUE THREAD

**Issue by Popax21** [Panakotta00/FicsIt-Networks#204]
Game version is the current Early Access branch (CL176027, SML v3.3.0), OS is Arch Linux. This happens with the latest version installed via SMM, `development` profile. Interestingly, if I manually install the latest version from SMR, it doesn't crash anymore, however it then crashes when starting a new game.

Crash log for the SMM setup:
```
LogNativeHookManager: Display: Attempting to resolve virtual function AFGGameMode::PostLogin. This adjustment: 0, virtual function table offset: 1888
LogNativeHookManager: Display: Successfully resolved virtual function thunk AFGGameMode::PostLogin at 000000007ABB7884 to function implementation at 0000000046E33100
LogNativeHookManager: Display: Hooking function AFGGameMode::PostLogin: Provided address: 000000007ABB7884, resolved address: 0000000046E33100
LogNativeHookManager: Display: Successfully hooked function AFGGameMode::PostLogin at 0000000046E33100
LogInit: Display: Starting Game.
LogAudio: Display: Audio Device unregistered from world 'None'.
LogAudio: Display: Audio Device (ID: 1) registered with world 'Map_MenuScene_Update_05'.
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Error: === Critical error: ===
LogWindows: Error:
LogWindows: Error: Fatal error!
LogWindows: Error:
LogWindows: Error: Unhandled Exception: EXCEPTION_ACCESS_VIOLATION 0x0000000101f
2a425
LogWindows: Error:
LogWindows: Error:
```

Crash log for the manual install setup:
```
LogContentRegistry: Display: Freezing content registry
LogNavigation: Error: Navmesh bounds are too large! Limiting requested tiles count (1481952) to: (1048576) for RecastNavMesh/Game/FactoryGame/Map/GameLevel01/Persistent_Level.Persistent_Level:PersistentLevel.RecastNavMesh-Default
LogStreaming: Error: Couldn't find file for package /FicsItNetworks/Components/ModularPanel/Modules/Module_RCO requested by async loading code. NameToLoad: /FicsItNetworks/Components/ModularPanel/Modules/Module_RCO
LogStreaming: Error: This will hitch streaming because it ends up searching the disk instead of finding the file in the pak file.
LogStreaming: Error: Found 0 dependent packages...
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Warning: SymLoadModuleExW. Error: 6
LogWindows: Error: === Critical error: ===
LogWindows: Error:
LogWindows: Error: Assertion failed: ModuleRCO [File:G:/Projects/Satisfactory/SatisfactoryModLoader/Plugins/FicsItNetworks/Source/FicsItNetworks/FicsItNetworksModule.cpp] [Line: 226]
LogWindows: Error:
LogWindows: Error:
LogWindows: Error:
LogWindows: Error: [Callstack] 0x000000007b0123be EntryPoint() []
LogWindows: Error: [Callstack] 0x000000018026e0c6 ReportAssert() []
LogWindows: Error: [Callstack] 0x0000000180270866 FWindowsErrorOutputDevice::Serialize() []
LogWindows: Error: [Callstack] 0x00000001801e93dd FOutputDevice::LogfImpl() []
LogWindows: Error: [Callstack] 0x000000018019fd3b AssertFailedImplV() []
LogWindows: Error: [Callstack] 0x00000001801a1900 FDebug::CheckVerifyFailedImpl() []
LogWindows: Error:
```

**Comment by Popax21** [comment]
**NOTES**
- I couldn't find a difference in the contents of the manual install folder and the SMM install folder
- I *could* get the game to launch the SMM version by renaming the folder, however it still crashes when starting a new game

**Comment by Panakotta00** [comment]
Linux is not supported for FIN because SF doesnt even support it, and mods for dedicated servers are also not supported

**Comment by Popax21** [comment]
With "on Linux" I mean using Proton/Wine.

**Comment by Panakotta00** [comment]
I know... Proton/Wine != Windows Runtime
Sry I simply dont have any time to even think about supporting Linux
feel free to create a PR tho

**Comment by Popax21** [comment]
Ah OK, as it seems to be related to `SymLoadModuleExW` (as this fails right before crashing), I'm gonna try to look into possible issues regarding that.
Also note that it worked perfectly before Update 5, so it isn't an inherit Proton issue.

**Comment by Popax21** [comment]
I figured it out: a wine version somewhere between version 6.19 (my system's) and 6.3 (first broken proton one) introduces a bug somewhere related to `SymLoadModuleExW`(am still investigating this) which causes it to fail and Satisfactory to subsequently crash. One workaround is to downgrade to a proton version before or at version 6.19.

**TLDR** Force Satisfactory to use Proton version `5.13-6` for now

**Comment by Deukhoofd** [comment]
As a note for future people: this can also be resolved by adding -d3d11 as a launch option.

**Comment by RenaKunisaki** [comment]
Adding `-d3d11` didn't help for me, but disabling some mods did. It still crashes sometimes, but it's stable once you get in-game. I noticed when it fails, the memory usage is 3.6GB just before crash, but when it loads OK, it's 3.0GB at the menu. I wonder if some 32-bit limit is to blame? (It does go over 5GB in-game though)
